### ISSUE TITLE
Fey Nightsight does not allow for crafting and reading in darkness

### ISSUE THREAD

**Issue by Sacrez** [CleverRaven/Cataclysm-DDA#13101]
Version: 0.C-4708-g5a939ff

I have fey nightsight and according to http://www.wiki.cataclysmdda.com/index.php?title=Fey_Nightsight " Allows reading and crafting without any source of light. ".

Proof: http://gyazo.com/17bd2354d2e6a25cd1216f1024bff90c

Left you can see my mutations and on the bottom right you can see I have attempted to both read and craft.

Steps to reproduce: 
1) Get Fey Nightsight
2) Attempt to read or/and craft

**Comment by DavidKeaton** [comment]
Correct me if I'm wrong, but wasn't this discussed? Not saying it should be closed or dismissed, I don't remember the outcome (if it even did happen).

**Comment by Barhandar** [comment]
>Correct me if I'm wrong, but wasn't this discussed?

Was, "nightvision doesn't give fine enough detail to read and craft".  
The problem is that Fey Nightsight's description is "You can see clearly, even in the darkest caves and under moonless skies." which kiiiinda implies it _should_ be possible to read and craft with it.

Also, before the lighting overhaul, Fey Nightsight had one less tile of visibility than Full Night Vision. I wonder if it persists.

**Comment by DavidKeaton** [comment]
Ah, okay.

**Comment by Sacrez** [comment]
@Barhandar It's the same as night vision bionic, so I assume it's also the same as full night vision.

**Comment by drbig** [comment]
> "You can see clearly, even in the darkest caves and under moonless skies."

Maybe just change description then?

**Comment by Barhandar** [comment]
>Maybe just change description then?

I'd rather have any NV that can compare to being under a clear night sky with a full moon "cannot see finer details, but can read and craft".

**Comment by kevingranade** [comment]
People can't generally read under bright moonlight either.  If you want to
read or craft, get some light.
Fey Nightvision and its description were added when many of the nightvision
types allowed reading, it needs to brought in line with the general rule
that nightvision does not allow reading or crafting.

In fact, now that this has my attention, the description is simply
nonsensical, no matter how good your vision is, you can't see in total
darkness.

**Comment by drbig** [comment]
> In fact, now that this has my attention, the description is simply
nonsensical, no matter how good your vision is, you can't see in total
darkness.

This then turns this into an easily-fixable bug.

**Comment by Barhandar** [comment]
>People can't generally read under bright moonlight either. 

I can read, with some issue, under full moon. People with good sight should be able to do the same. 
>In fact, now that this has my attention, the description is simply nonsensical, no matter how good your vision is, you can't see in total darkness.

The entire Elf-a line's decription gives me vibes of being full of poetic exaggeration. Which doesn't have to make sense.
Also, it could either stretch into infrared, or actually work as its own (low-visibility - such as the mentioned infrared or ultraviolet) light source - the eyes need to be mutated for it, after all.

**Comment by kevingranade** [comment]
If it's poetic exaggeration we're good, because the description doesn't
have to match how it works, right?

Infrared or ultraviolet aren't going to help you at the bottom of a mine,
everything is the same temperature so there wouldn't be any fluctuations in
order to pick up details, and I'm not sure where the ultraviolet would be
coming from.

**Comment by Sacrez** [comment]
Perhaps simply change the description to something along the lines of:
Enables great sight in low light conditions. (Maybe add "but does not allow for fine tasks such as crafting and reading")

If we're to talk about the ideas behind it all then:

If we're to consider physics in the idea of night vision we're going to run into a lot of issues I believe. At the bottom of a mine there would no sight whatsoever as there would be no light. But if you're in a house and there's for example a light shining from the outside, the light will bounce around on every object and it will make everything brighter.
We also have to deal with the fact that we don't know how having a biological "night vision" would be like so our closest equivalent is something from a machine. Perhaps this is a good avenue for adding in something else to the game, such as FLIR https://www.youtube.com/watch?v=rAvnMYqj2c0 .

Bottom line: In my opinion there ought to be some kind of a compromise and allow for night vision in all situations of low/no light or otherwise work in broad sweeps, if in an underground place, no night vision, if in any above ground situation sure night vision.

But yeah I'm not the most informed person as to the inner workings.

**Comment by DavidKeaton** [comment]
FLIR is heat signature, and UV light could be a byproduct of radioactivity in minerals in the earth iirc. However, the amplification of infrared light does need something to have _SOME_ light to begin with. With the case of IR goggles, there is usually an IR light that provides it if there is none, however with this mutation, your eyes are merely able to see IR, not able to produce some to illuminate as well.

**Comment by macrosblackd** [comment]
It should be noted that cheap night vision devices (toys, security cameras, etc) use an IR emitter and a camera. This DOES provide enough light to read with.

**Comment by DavidKeaton** [comment]
Even military grade hardware has an IR emitter.

**Comment by drbig** [comment]
> use an IR emitter and a camera

Find me a biological example of built-in IR emitter for the purpose of 'hunting at night'. Or any biological directional IR emitter for that matter. And no, body heat won't work.

**Comment by DavidKeaton** [comment]
I don't think he was speaking of biological, nor was I.

**Comment by drbig** [comment]
Sure. Fine for a CBM, but mutations are biological...

**Comment by macrosblackd** [comment]
And both are in the realm of fantasy/scifi if not HIGH fantasy/scifi. What's wrong with handwaving on the biological side?

**Comment by DavidKeaton** [comment]
Most, if not all, programming editors can do that.

**Comment by DavidKeaton** [comment]
Well given so many other aspects are reality related, and mutations give animal or chemical changes that are relatable.

**Comment by sammyjudo** [comment]
2 things first some of the mutations are somewhat impossible such as amorphous body (by the way find me a natural example of that). Second there a mutations interfacing with technology like the prototypes immunities. So I don't  see why crafting in the darks is preposterous with super intelligent blobs that can creat eyes and ears at will and people with antiseptic blood.

**Comment by sammyjudo** [comment]
So to resolve this you could make this a mod when creating a world

**Comment by drbig** [comment]
@sammyjudo Yeah. It's a game indeed :) I'm not against read+craft enabling nightvision per se.

For this one here though the *immediate* fix is to change the description.

I would be most for a NV CBM that does include IR emitter though.

**Comment by macrosblackd** [comment]
A biological example of something that isn't too far off from what we're discussing is the angler fish. The source of the luminescence is a bacteria. It shouldn't be difficult to imagine a scenario where there is some bacteria that emits near infrared light that can be controlled some how similar to the angler fish's light.

**Comment by sammyjudo** [comment]
just make an optional mod

**Comment by kevingranade** [comment]
On Jul 30, 2015 1:37 PM, "Justin" <notifications@github.com> wrote:
>
> A biological example of something that isn't too far off from what we're
discussing is the angler fish. The source of the luminescence is a
bacteria. It shouldn't be difficult to imagine a scenario where there is
some bacteria that emits near infrared light that can be controlled some
how similar to the angler fish's light.

Go right ahead... and have a light source permanently attached to your
forehead, this does not sound like a good tradeoff :)
Regardless, you're getting into a totally different thing now, my issue is
that biological night vision simply is not well suited to detail-oriented
tasks, if you can add some light, infrared or ultraviolet or whatever,
we're no longer talking about the same thing.
The current reading/crafting situation is that if you have the dimmest
light source in the game on your person, you can use it to read and craft,
so if someone added a 'bioluminescent lure' mutation, it would certainly
qualify, but as I pointed out, it will also count as a light source,
meaning monsters can see you.

**Comment by DavidKeaton** [comment]
Wait we were discussing about why they could have a light emitter? Oh man, as far as a mut goes, no way. If this didn't come across from my statements, my b.

**Comment by i2amroy** [comment]
Honestly here's what I think we need to do.
1) Remove the nonsensical cap on darkness at the "I can see one tile" range. A player in totally pitch darkness *should* be acting the same as one that is totally blind, not have some magical "I can see 1 tile away despite it being pitch black" vision.
2) Normal darkness would still only go down to "I can see one tile" range. The only exceptions would be places very far from any source of light, or those that are underground.
3) **All** NV stops working if the light level hits zero. This even includes technological NV sources, since even those will fail in absolute darkness.
3) Night vision will let you craft in lower light levels, because it does in real life. However, since all NV stops working at a light level of 0, you can't craft at that level, regardless of how much NV you have available.
4) Balance Fey vision as the "you can see in any light level brighter than 0" NV mutation. Either keep the description as poetic prose, or rewrite it slightly (attempting to keep the feel).

By doing this we get more realistic NV, the player acts more realistically in pitch blackness, and we remove a silly hardcoded limit and swap it out for a more sensible one.

**Comment by Rivet-the-Zombie** [comment]
>A player in totally pitch darkness should be acting the same as one that is totally blind, not have some magical "I can see 1 tile away despite it being pitch black" vision

We need it because we can't otherwise simulate stuff like feeling your way around in the dark or spatial senses.

>All NV stops working if the light level hits 0. This even includes technological NV sources, since even those will and should fail in absolute darkness.

It seems like this would make the night vision traits useless in most circumstances. Besides that, a lot of those fancy technological night vision systems have emitters that can provide illumination in whatever spectrum of light that they're attuned to.

I'd rather we stay with the current behavior here.

**Comment by illi-kun** [comment]
> I'd rather we stay with the current behavior here.

+1

**Comment by kevingranade** [comment]
I'm on board with the first three points in principle, but we'd need
features to let you navigate a little bit, even if it's just identifying
things when you try to walk into them.  The problem is that kind of system
assumes your perception of at least walls is persistent, but in dda it is
not, I don't have a solution for adding that kind of feature.
The other problem is implementing 'total darkness' even just underground
means all nightvision (other than IR-based vision with IR illumination) is
not going to work at all there, which seems overly harsh and makes these
abilities almost completely pointless.
One possibility is having some degree of lighting in at least some of these
areas, such as emergency lighting in human-created facilities, which is the
majority of the interesting underground area.  I think total darkness in
L2+ of caves might be ok. (Assuming some light coming from whatever
entrance there is, or something more sophisticated if desired)
I'm against crafting/reading with NV because it simply doesn't work like
that, night vision is optimised for being able to make out coarse details
in low-light conditions, but it sacrifices resolution to make it happen.
Flagging crafting recipes that don't require detail vision to perform, or
vice versa if not needing detail vision is the majority would be possible,
but we'd probably get even more complaints that it's 'inconsistent' if we
did that.

**Comment by Sacrez** [comment]
Overall there have been some interesting points made here. 
A system of detecting our environment without seeing (echolocation? Maybe new mutation tree "Bat"?)  would definitely be interesting but it does sound a little bit far ahead.
The other suggestions are not without their merit but they may create more inconvenience than it would be justified by them being more immersive/realistic.
I believe overall the system as it is right now works just fine, at the very least until the introduction of new/different mechanics. And maybe I'm wrong here but if a new mechanic in regards to this would be implemented I'd have it answer all possibilities rather than going like "Oh, this very specific set of circumstances has been reached, I will act different now". This only unnecessarily lengthens descriptions and gives the game more micro management.
"You can see in the dark but you can't craft or read" Sounds just about fair enough as of right now.

**Comment by Chase-san** [comment]
I wouldn't mind an activatable mutation that produces some kind of light or even a non-activatable one, or both actually, the non-activatable one progressing to the activatable one.
